WordPress/gutenberg
# Elevation: Convert component to TypeScript

## What?
Converts the `Elevation` component to TypeScript.

## Why?
Part of the @wordpress/components's TypeScript migration (https://github.com/WordPress/gutenberg/issues/35744).

## How?
- Refactors the current `Elevation` to TypeScript.
- Refactors the test to TypeScript.
- Refactors storybook to TypeScript and controls. With controls the storybook can be simplified.

## Testing Instructions
1. Run npm run dev and ensure no typing errors
2. Fire up the Storybook and confirm the `Elevation` continues to function as expected
